### ISSUE TITLE
Sub topic with wildcards,can't get the actual topic what publisher sent when dapr callback.The callback topic the same as what I sub,still remain wildcards.

### ISSUE THREAD

**Issue by OyuO** [dapr/dapr#4826]
## In what area(s)?

/area runtime

> /area operator

> /area placement

> /area docs

> /area test-and-release

## What version of Dapr?

1.7.4
> 1.1.x
> 1.0.x
> edge: output of `git describe --dirty`

## Expected Behavior

If I sub *.weve.event, I should receive  topics without wildcards,such as  123.weve.event,hello.wev.event

## Actual Behavior

But when callback, I  get topic from the grpc request, I get "*.weve.event" actually,I is amazing,I can't get the real topic. I need get info from topic.

## Steps to Reproduce the Problem
1. Pubsub component: nats
2. Sub topic "*.weve.evnet" by dapr ,set the rawPayload = true,and implement the onEvent() by grpc,print the topic.
3.  Pub hello to topic hello.weve.event
## Release Note

RELEASE NOTE:

**Comment by yaron2** [comment]
Must you use `rawPayload=true`?

**Comment by OyuO** [comment]
Yes! Because I need to communicat with other system,they don't use dapr.
